### DISCUSSION TITLE
Firebase v9 API support

### DISCUSSION THREAD

**Discussion by scblaze** [invertase/react-native-firebase#5838] [General]
We have code shared between a project that uses react-native-firebase and a project using the official Firebase SDK's.

We would like to update this code to use the v9 Firebase API.

Could react-native-firebase be updated to support this new API?

**Comment by mikehardy** [comment]
Hi there - it's sort of in progress, but waiting on a new test harness before we can write tests for it which is a pre-requisite to releasing it so it has high quality - https://github.com/invertase/react-native-firebase/commits/%40invertase/next

Which is a detailed way of saying: there are definitely plans, but it will be a process that takes quite some time unfortunately. If you need a solution immediately I think the compat style of usage or staying on older JS sdk is it for the moment

**Reply by levino** [reply]
Looks like this has been more or less abandoned. Any update on the matter?

**Reply by mikehardy** [reply]
Just lack of time at the moment. I've got a basic next generation test framework up and running (https://github.com/invertase/jet/tree/%40mikehardy/jet-next / https://github.com/invertase/react-native-google-mobile-ads/tree/main/RNGoogleMobileAdsExample) but it's not quite complete. That second link is interesting in that react-native-google-mobile-ads took a lot of my personal time which then wasn't available for a JS SDK v9 API project here :shrug: - but it's a great module and was really needed by many. I just finished digesting the Google I/O 2022 breaking changes releases (https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#1500-2022-06-20) - they were a lot more time-consuming then I thought

Unfortunately it's about to be summer and I'm a regular human like most, with a kid that's about to be out of school and family to visit so it's likely to stay asleep until the Fall - but it's definitely on the list of things to do.

I wish I had 48hrs of time for every 24hr day but in the meantime we do and I do appreciate the patience

**Reply by yairopro** [reply]
@mikehardy You do great. And your family is the most important thing it should be for you. Have a great summer!😎

**Reply by levino** [reply]
I did not mean to be pushy about this of course. I understand that this is not something that your are being paid to do or promised on a delivery date and I would not think about being demanding about the implementation. When it happens, it happens. I did not want to complain that this is not done but just know the status of this. I hope I have not created any bad feelings here. I appreciate all the important work you are doing and also this very useful library right here.

**Reply by mikehardy** [reply]
Oh no not at all! Actually on behalf of Invertase this is a module I *am* paid to work on, there just aren't enough hours in the day - typical programming problem. For most of the other modules I work on it's volunteer stuff but for anything in Invertase org (and related social-authentication libraries like fbsdk-next etc) it's actually a job :-). No worries regardless. Cheers!

**Reply by hirvesh** [reply]
Hey @mikehardy just wanted to know if there's anything happening on this front :-)

**Reply by mikehardy** [reply]
Nothing yet, sorry - finally recovering from a huge summer travel stint and thinking of next steps for this repo, it's got to be considered co-#1 along with switching to Typescript so that we can more easily support New Architecture. Can't delay it forever, that's a true statement. I'm still really really hesitant to give any sort of deadline though, those never work out in an open source context
